src/provider/rpc.ts

```diff
@@ -90,14 +90,17 @@ export class RpcProvider implements ProviderInterface {
   }
 
   public async getBlock(blockIdentifier: BlockIdentifier = 'pending'): Promise<GetBlockResponse> {
-    const method =
-      typeof blockIdentifier === 'string' && isHex(blockIdentifier)
-        ? 'starknet_getBlockByHash'
-        : 'starknet_getBlockByNumber';
-
-    return this.fetchEndpoint(method, [blockIdentifier]).then(
-      this.responseParser.parseGetBlockResponse
-    );
+    const method = 'starknet_getBlockWithTxHashes';
+    if (typeof blockIdentifier === 'string' && isHex(blockIdentifier)) {
+      return this.fetchEndpoint(method, [{ block_hash: blockIdentifier }]).then(
+        this.responseParser.parseGetBlockResponse
+      );
+    }
+    else {
+      return this.fetchEndpoint(method, [{ block_number: blockIdentifier }]).then(
+        this.responseParser.parseGetBlockResponse
+      );
+    }
```

Comment:
by RPC 0.1.0 
method 'starknet_getBlockWithTxHashes'
has parameter 'block_id' that is hash, number or block id ?

https://github.com/starkware-libs/starknet-specs/blob/v0.1.0/api/starknet_api_openrpc.json

Reply:
blockIdentifier can now be string, number or tag
https://github.com/starkware-libs/starknet-specs/blob/30e5bafcda60c31b5fb4021b4f5ddcfc18d2ff7d/api/starknet_api_openrpc.json#L767